go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -363,12 +329,18 @@ func (a *PodAllocator) allocatePodOnNAD(pod *corev1.Pod, nad string, network *ne
 		idAllocator = a.idAllocator.ForName(name)
 	}
 
-	// don't reallocate to new IPs if currently annotated IPs fail to alloccate
+	// don't reallocate to new IPs if currently annotated IPs fail to allocate
 	reallocate := false
 	networkRole, err := a.GetNetworkRole(pod)
 	if err != nil {
 		return err
 	}
+
+	if networkRole == types.NetworkRoleNone {
+		// pod not on this controller, nothing to do
+		return nil
+	}
```

Comment:
Could this happen earlier? Like the very first thing to check when handling the event?

Reply:
Noting that there seems to be some overlap with what is done in `GetPodNADToNetworkMappingWithActiveNetwork` which also checks if the pod is on the network. Can be improved in the future.

Reply:
yes i didnt look to see if it could be moved up earlier in the functions as an optimization. if there are other changes needed to this PR i will take a look at that possibility.